src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,64 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionRequestDto;
+import mate.academy.spring.dto.MovieSessionResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.DtoMapper;
+import mate.academy.spring.service.MovieSessionService;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final DtoMapper<MovieSession,
+            MovieSessionRequestDto, MovieSessionResponseDto> dtoMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  DtoMapper<MovieSession, MovieSessionRequestDto,
+                                          MovieSessionResponseDto> mapper) {
+        this.movieSessionService = movieSessionService;
+        this.dtoMapper = mapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto create(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        return dtoMapper.toDto(movieSessionService.add(dtoMapper.toModel(movieSessionRequestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSessions(
+            @RequestParam Long movieId,
+            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date).stream()
+                .map(dtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("/{id}")
+    public MovieSessionResponseDto update(
+            @PathVariable Long id, @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```

Comment:
Let's keep this formating style. This is more readable and easier to update
```suggestion
            @PathVariable Long id, 
            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```